pycbc/events/stat.py

```diff
@@ -1758,6 +1757,155 @@ def coinc_lim_for_thresh(self, s, thresh, limifo,
         return loglr
 
 
+
+class ExpFitFgBgKDEStatistic(ExpFitFgBgNormStatistic):
+    """
+    The ExpFitFgBgNormStatistic with a mass  and spin weighting factor determined by KDE
+
+    This is the same as the ExpFitFgBgNormStatistic except the likelihood
+    is multiplied by the ratio of signal KDE to template KDE over template params.
+    """
+
+    def __init__(self, sngl_ranking, files=None, ifos=None, **kwargs):
+        """
+        Create a statistic class instance
+
+        Parameters
+        ----------
+        sngl_ranking: str
+            The name of the ranking to use for the single-detector triggers.
+        files: list of strs, needed here
+            A list containing the filenames of hdf format files used to help
+            construct the coincident statistics. The files must have a 'stat'
+            attribute which is used to associate them with the appropriate
+            statistic class.
+        ifos: list of strs, not used here
+            The list of detector names
+        """
+        ExpFitFgBgNormStatistic.__init__(self, sngl_ranking, files=files,
+                                         ifos=ifos, **kwargs)
+        parsed_attrs = [f.split('-') for f in self.files.keys()]
```

Comment:
perhaps add a comment here to say what are the exact names of the attributes you are looking for?